packages/angular/cli/models/analytics-collector.ts

```diff
@@ -0,0 +1,312 @@
+    this.parameters['cd' + analytics.NgCliAnalyticsDimensions.CpuSpeed] = Math.floor(os.cpus()[0].speed);
+    this.parameters['cd' + analytics.NgCliAnalyticsDimensions.RamInGigabytes] = Math.round(os.totalmem() / (1024 * 1024 * 1024));
+    this.parameters['cd' + analytics.NgCliAnalyticsDimensions.NodeVersion] = nodeVersion;
+  }
+
+  event(ec: string, ea: string, options: analytics.EventOptions = {}): void {
+    const { label: el, value: ev, metrics, dimensions } = options;
+    this.addToQueue('event', { ec, ea, el, ev, metrics, dimensions });
+  }
+
+  pageview(dp: string, options: analytics.PageviewOptions = {}): void {
+    const { hostname: dh, title: dt, metrics, dimensions } = options;
+    this.addToQueue('pageview', { dp, dh, dt, metrics, dimensions });
+  }
+
+  timing(utc: string, utv: string, utt: string | number, options: analytics.TimingOptions = {}): void {
+    const { label: utl, metrics, dimensions } = options;
+    this.addToQueue('timing', { utc, utv, utt, utl, metrics, dimensions });
+  }
+
+  screenview(cd: string, an: string, options: analytics.ScreenviewOptions = {}): void {
+    const { appVersion: av, appId: aid, appInstallerId: aiid, metrics, dimensions } = options;
+    this.addToQueue('screenview', { cd, an, av, aid, aiid, metrics, dimensions });
+  }
+
+  async flush(): Promise<void> {
+    const pending = this.trackingEventsQueue.length;
+    this.analyticsLogDebug(`flush queue size: ${pending}`);
+
+    if (!pending) {
+      return;
+    }
+
+    // The below is needed so that if flush is called multiple times.
+    // we don't report the same event multiple times.
+    const pendingTrackingEvents = [...this.trackingEventsQueue];
+    this.trackingEventsQueue = [];
+
+    await this.send(pendingTrackingEvents);
+  }
+
+  private addToQueue(eventType: 'event', parameters: EventParameters): void;
+  private addToQueue(eventType: 'pageview', parameters: PageviewParameters): void;
+  private addToQueue(eventType: 'timing', parameters: TimingParameters): void;
+  private addToQueue(eventType: 'screenview', parameters: ScreenviewParameters): void;
+  private addToQueue(eventType: 'event' | 'pageview' | 'timing' | 'screenview', parameters: BaseParameters): void {
+    const { metrics, dimensions, ...restParameters } = parameters;
+    const data = {
+      ...this.parameters,
+      ...restParameters,
+      ...this.customVariables({ metrics, dimensions }),
+      t: eventType,
+    };
+
+    this.analyticsLogDebug('add event to queue: %j', data);
+    this.trackingEventsQueue.push(data);
+  }
+
+  private async send(data: Record<string, string | number | boolean>[]): Promise<void> {
+    this.analyticsLogDebug('send event: %j', data);
+
+    return new Promise<void>((resolve, reject) => {
+      const request = https.request({
+        protocol: 'https:',
+        host: 'www.google-analytics.com',
+        method: 'POST',
+        path: data.length > 1 ? '/batch' : '/collect',
+      }, response => {
```

Comment:
Maybe we don't need this whole response handler if we never care about the response.
According to the [documentation](https://nodejs.org/docs/latest-v12.x/api/http.html#http_class_http_clientrequest),
> If no 'response' handler is added, then the response will be entirely discarded. 

If we do this, then we can probably resolve the Promise as soon as the request is sent.

Reply:
Good point